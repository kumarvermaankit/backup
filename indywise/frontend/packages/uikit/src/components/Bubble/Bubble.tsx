import * as React from 'react';
import styled from 'styled-components';

export interface IBubbleProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  style?: any;
  color?: string;
  size?: string;
}

export const Bubble: React.FC<IBubbleProps> = (props) => {
  return (
    <>
      <BubbleBody
        size={props.size}
        color={props.color}
        style={props.style || {}}
      />
    </>
  );
};

const BubbleBody = styled.div<{
  size: string | undefined;
  color: string | undefined;
}>`
  height: ${(props) => props.size || '100px'};
  width: ${(props) => props.size || '100px'};
  border-radius: 50%;
  position: absolute;
  z-index: -1;
  background: ${(props) => props.color || '#DDC2E4'};
`;
