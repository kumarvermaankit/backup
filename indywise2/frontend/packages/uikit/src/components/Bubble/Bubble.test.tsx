import * as React from 'react';
import { create } from 'react-test-renderer';

import { Bubble } from './Bubble';

describe('Bubble', () => {
  it('Size', () => {
    const component = create(<Bubble size="100px" />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Color', () => {
    const component = create(<Bubble color="#222" />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Style', () => {
    const component = create(<Bubble style={{}} />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
