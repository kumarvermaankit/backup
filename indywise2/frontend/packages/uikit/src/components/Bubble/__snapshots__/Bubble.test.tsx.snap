// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble Color 1`] = `
<div
  className=""
  color="#222"
  style={Object {}}
/>
`;

exports[`Bubble Size 1`] = `
<div
  className=""
  size="100px"
  style={Object {}}
/>
`;

exports[`Bubble Style 1`] = `
<div
  className=""
  style={Object {}}
/>
`;
